src/zenml/artifacts/utils.py

```diff
@@ -763,7 +767,19 @@ def _load_file_from_artifact_store(
     """
     try:
         with artifact_store.open(uri, mode) as text_file:
-            return text_file.read()
+            if offset < 0:
+                # If the offset is negative, we seek backwards from the end of
+                # the file
+                try:
+                    text_file.seek(offset, os.SEEK_END)
+                except OSError:
+                    # The negative offset was too large for the file, we seek
+                    # to the start of the file
+                    text_file.seek(0, os.SEEK_SET)
+            elif offset > 0:
+                text_file.seek(offset, os.SEEK_SET)
+
+            return text_file.read(length)
```

Comment:
This method is also used to load visualization files in the dashboard. You should test that still works properly with this (i.e. not sure what happens if length is None here)

Reply:
`read(None)` should be equivalent to just `read()` which was the previous behaviour